uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -69,24 +69,29 @@ def full_auto_attach(options: FullAutoAttachOptions) -> FullAutoAttachResult:
 
 
 def _full_auto_attach(
-    options: FullAutoAttachOptions, cfg: UAConfig
+    options: FullAutoAttachOptions,
+    cfg: UAConfig,
+    *,
+    mode: event_logger.EventLoggerMode = event_logger.EventLoggerMode.JSON
 ) -> FullAutoAttachResult:
     try:
         with lock.SpinLock(
             cfg=cfg,
             lock_holder="pro.api.u.pro.attach.auto.full_auto_attach.v1",
         ):
-            ret = _full_auto_attach_in_lock(options, cfg)
+            ret = _full_auto_attach_in_lock(options, cfg, mode=mode)
     except Exception as e:
         lock.clear_lock_file_if_present()
         raise e
     return ret
 
 
 def _full_auto_attach_in_lock(
-    options: FullAutoAttachOptions, cfg: UAConfig
+    options: FullAutoAttachOptions,
+    cfg: UAConfig,
+    mode: event_logger.EventLoggerMode,
 ) -> FullAutoAttachResult:
-    event.set_event_mode(event_logger.EventLoggerMode.JSON)
+    event.set_event_mode(mode)
```

Comment:
I just want to acknowledge this change. I don't like that we have to do this, but I think we're stuck doing this until more functions (i.e. enable and attach) are migrated to the API.

Reply:
Being discussed here: https://github.com/canonical/ubuntu-advantage-client/pull/2297#discussion_r1010438664